### PR TITLE
Adds fzero keymap for GMMK Pro ANSI rev1

### PR DESCRIPTION
**Pull request by fzero**
## Description

Adds fzero keymap for GMMK Pro ANSI rev1

## Types of Changes

- [ ] Core
- [ ] Bugfix
- [ ] New feature
- [ ] Enhancement/optimization
- [ ] Keyboard (addition or update)
- [x] Keymap/layout/userspace (addition or update)
- [ ] Documentation

## Checklist

- [x] My code follows the code style of this project: [**C**](https://docs.qmk.fm/#/coding_conventions_c), [**Python**](https://docs.qmk.fm/#/coding_conventions_python)
- [x] I have read the [**PR Checklist** document](https://docs.qmk.fm/#/pr_checklist) and have made the appropriate changes.
- [ ] My change requires a change to the documentation.
- [x] I have updated the documentation accordingly.
- [x] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [ ] I have added tests to cover my changes.
- [x] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

File: keyboards/gmmk/pro/rev1/ansi/keymaps/fzero/README.md [hunk 1]
```diff
@@ -0,0 +1,50 @@
+# fzero's GMMK Pro layout
```
Comment (drashna) [inline keyboards/gmmk/pro/rev1/ansi/keymaps/fzero/README.md:1]: Please rename the file, so that it's lower case.

**Review by drashna — CHANGES_REQUESTED** [review]

File: keyboards/gmmk/pro/rev1/ansi/keymaps/fzero/README.md (hunk 1 above)
Comment (drashna) [inline keyboards/gmmk/pro/rev1/ansi/keymaps/fzero/README.md:1]: Please rename the file to be lower case.

**Review by drashna — CHANGES_REQUESTED** [review]

**Review by drashna — APPROVED** [review]

**Review by fauxpark — APPROVED** [review]

File: keyboards/gmmk/pro/rev1/ansi/keymaps/fzero/keymap.c [hunk 2]
```diff
@@ -0,0 +1,391 @@
...
+        break;
+    #endif // RGB_MATRIX_ENABLE
+
+    }
+    return true;
+};
+
+// --- RGB SUPPORT FUNCTIONS ---------------------------------------------------
+#ifdef RGB_MATRIX_ENABLE
+void rgb_matrix_indicators_user() {
```
Comment (drashna) [inline keyboards/gmmk/pro/rev1/ansi/keymaps/fzero/keymap.c:310]:
```suggestion
bool rgb_matrix_indicators_user(void) {
```

File: keyboards/gmmk/pro/rev1/ansi/keymaps/fzero/keymap.c [hunk 3]
```diff
@@ -0,0 +1,391 @@
...
+            }
+        }
+    }
+    if (rgb_matrix_get_flags() == LED_FLAG_CAPS) {
+        rgb_matrix_set_color_all(0x0, 0x0, 0x0);
+    }
+    if (host_keyboard_led_state().caps_lock) {
+        set_rgb_caps_leds();
+    }
+}
```
Comment (drashna) [inline keyboards/gmmk/pro/rev1/ansi/keymaps/fzero/keymap.c:346]:
```suggestion
    return false;
}
```

**Review by drashna — CHANGES_REQUESTED** [review]

**Review by drashna — APPROVED** [review]

**Review by drashna — CHANGES_REQUESTED** [review]

**Pull request closed without merging**